src/speckle_automate/automation_context.py

```diff
@@ -415,19 +444,40 @@ def attach_result_to_objects(
         self,
         level: ObjectResultLevel,
         category: str,
-        object_ids: Union[str, List[str]],
+        affected_objects: Union[Base, List[Base]],
         message: Optional[str] = None,
         metadata: Optional[Dict[str, Any]] = None,
         visual_overrides: Optional[Dict[str, Any]] = None,
     ) -> None:
-        if isinstance(object_ids, list):
-            if len(object_ids) < 1:
+        """Add a new result case to the run results.
+
+        Args:
+            level: Result level.
+            category (str): A short tag for the event type.
+            affected_objects (Union[Base, List[Base]]): A single object or a list of
+                objects that are causing the info case.
+            message (Optional[str]): Optional message.
+            metadata: User provided metadata key value pairs
+            visual_overrides: Case specific 3D visual overrides.
+        """
+        # validate that the Base.id is not None. If its a None, throw an Exception
+        if isinstance(affected_objects, list):
+            if len(affected_objects) < 1:
                 raise ValueError(
                     f"Need atleast one object_id to report a(n) {level.value.upper()}"
```

Comment:
This message still refers to Ids,
Probably fine tbh, but I'll make sure the C# version be `$"Need at least one affected object to report a(n) {level}"`